qltui.py

```diff
@@ -726,18 +660,27 @@ def hook(ql):
             hook_dictionary["end"] = end
 
         if user_data:
-            args.append(user_data)
             hook_dictionary["user_data"] = user_data
 
-        if hook_type in ['hook_code', 'hook_block']:
-            operation += '_4_args'
+        getattr(ql, hook_type)(getattr(Callback_Functions, operation), *args, user_data=user_data)
 
-        elif hook_type in ['hook_mem_read', 'hook_mem_write']:
-            operation += '_6_args'
+    return hook_dictionary
 
-        else:
-            operation += '_2_args'
 
-        getattr(ql, hook_type)(getattr(Callback_Functions, operation), *args)
+def transform_syscalls(syscalls, keys=["address"], func=lambda x: hex(x)):
+    for i in syscalls:
+        try:
+            if type(i) == type({}):
```

Comment:
Hi @elicn, `Iterable` is also picking up string that is not giving the correct result. I clubbed it using or to have just two conditionals.

Reply:
Well, that wasn't for filtering bad arguments but for generalizing the condition.
If you go with the `isinstance` and explicit types, then you could write `isinstance(x, (type1, type2, type3)` instead of `or`-ing them. Anyway, I am fine with this either.